pkg/api/gitops/v1alpha1/application.go

```diff
@@ -236,6 +236,32 @@ type KustomizationSpec struct {
 	Wait bool `json:"wait,omitempty"`
 }
 
+// FluxApplicationStatus represent the status of a FluxApp
+type FluxApplicationStatus struct {
+	// HelmReleaseStatus is a list of HelmRelease status for each HelmRelease
+	// that belongs to this FluxApp
+	HelmReleaseStatus []*HelmReleaseStatus `json:"helmReleaseStatus,omitempty"`
```

Comment:
Here, I prefer to use a `map[string]*HelmReleaseStatus`,so that we don't have to traverse every time when we want to to obtain specific object

Reply:
What is the key?

Reply:
> Here, I prefer to use a `map[string]*HelmReleaseStatus`,so that we don't have to traverse every time when we want to to obtain specific object

Yes. I think you are right. I actually use `map[string]*helmv2.HelmReleaseStatus` before but I don't know why it couldn't generate deepcopy method. It work well now and make it easy to use in the later status-controller.

Reply:
> What is the key?

the key should be the generated name of `hr` or`kustomize`

Reply:
Is that possible to give an example of the status? I cannot imagine what the data looks like.

Reply:
### a app that has a successfully reconciled Kustomization and a failed Kustomization.
```yaml
status:
    fluxApp:
      kustomizationStatus:
        # kustomization name
        default:
          conditions:
            - lastTransitionTime: '2022-09-06T04:28:48Z'
              message: >-
                kustomization path not found: stat
                /tmp/kustomization-3289914681/chengleqi: no such file or directory
              reason: ArtifactFailed
              status: 'False'
              type: Ready
          lastAttemptedRevision: master/4b8497c8c3dc8c7ad5c9cb66160dbd3bcc1cfd4f
          observedGeneration: 1
        # kustomization name
        my-project:
          conditions:
            - lastTransitionTime: '2022-09-06T04:29:51Z'
              message: 'Applied revision: master/4b8497c8c3dc8c7ad5c9cb66160dbd3bcc1cfd4f'
              reason: ReconciliationSucceeded
              status: 'True'
              type: Ready
          inventory:
            entries:
              - id: my-project_nginx-svc__Service
                v: v1
              - id: my-project_nginx-deployment_apps_Deployment
                v: v1
          lastAppliedRevision: master/4b8497c8c3dc8c7ad5c9cb66160dbd3bcc1cfd4f
          lastAttemptedRevision: master/4b8497c8c3dc8c7ad5c9cb66160dbd3bcc1cfd4f
          observedGeneration: 1
```

### a app that has a HelmRelease successfully reconciled.
```yaml
fluxApp:
    helmReleaseStatus:
      # hr name
      my-project:
        conditions:
          - lastTransitionTime: '2022-09-06T04:34:05Z'
            message: Release reconciliation succeeded
            reason: ReconciliationSucceeded
            status: 'True'
            type: Ready
          - lastTransitionTime: '2022-09-06T04:34:05Z'
            message: Helm install succeeded
            reason: InstallSucceeded
            status: 'True'
            type: Released
        helmChart: my-devops-project5s2r8/my-devops-project5s2r8-test-fluxcdk9b2r
        lastAppliedRevision: 0.1.0+1
        lastAttemptedRevision: 0.1.0+1
        lastAttemptedValuesChecksum: da39a3ee5e6b4b0d3255bfef95601890afd80709
        lastReleaseRevision: 1
        observedGeneration: 1
```